service/controller/clusterapi/cluster.go

```diff
@@ -0,0 +1,115 @@
+package clusterapi
+
+import (
+	clusterv1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/cluster/v1alpha1"
+	"github.com/giantswarm/apiextensions/pkg/clientset/versioned"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/giantswarm/operatorkit/client/k8scrdclient"
+	"github.com/giantswarm/operatorkit/controller"
+	"github.com/giantswarm/operatorkit/informer"
+	corev1 "k8s.io/api/core/v1"
+	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	"sigs.k8s.io/cluster-api/pkg/client/clientset_generated/clientset"
+
+	"github.com/giantswarm/cluster-operator/pkg/cluster"
+	v16 "github.com/giantswarm/cluster-operator/service/controller/clusterapi/v16"
+)
+
+// ClusterConfig contains necessary dependencies and settings for
+// Cluster API's Cluster CRD controller implementation.
+type ClusterConfig struct {
+	BaseClusterConfig *cluster.Config
+	CMAClient         clientset.Interface
+	G8sClient         versioned.Interface
+	K8sExtClient      apiextensionsclient.Interface
+	Logger            micrologger.Logger
+
+	ProjectName string
+}
+
+type Cluster struct {
+	*controller.Controller
+}
+
+// NewCluster returns a configured AWSClusterConfig controller implementation.
+func NewCluster(config ClusterConfig) (*Cluster, error) {
+	if config.G8sClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.G8sClient must not be empty", config)
+	}
+
+	var err error
+
+	var crdClient *k8scrdclient.CRDClient
+	{
+		c := k8scrdclient.Config{
+			K8sExtClient: config.K8sExtClient,
+			Logger:       config.Logger,
+		}
+
+		crdClient, err = k8scrdclient.New(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	var newInformer *informer.Informer
+	{
+		c := informer.Config{
+			Logger:  config.Logger,
+			Watcher: config.CMAClient.ClusterV1alpha1().Clusters(corev1.NamespaceAll),
+
+			RateWait:     informer.DefaultRateWait,
+			ResyncPeriod: informer.DefaultResyncPeriod,
+		}
+
+		newInformer, err = informer.New(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	var resourceSetV16 *controller.ResourceSet
+	{
+		c := v16.ResourceSetConfig{
```

Comment:
I matched this version with other controllers' version bundle versions so that this is consistent with current practices.